identity-service/src/routes/cognito.js

```diff
@@ -56,15 +56,47 @@ module.exports = function (app) {
     const { id: sessionId, customer_reference: handle, status } = data
 
     try {
+      // check that this identity has not already been used by another account before proceeding to save the score
+      let shouldFail = false
+      if (status === 'success') {
+        const baseUrl = config.get('cognitoBaseUrl')
+        const path = `/flow_sessions/${sessionId}`
+        const method = 'GET'
+        const body = ''
+        const headers = createCognitoHeaders({ path, method, body })
+        const url = `${baseUrl}${path}`
+        const flowSessionResponse = await axios({
+          adapter: axiosHttpAdapter,
+          url,
+          method,
+          headers
+        })
+        const identityObj = flowSessionResponse.data.user.id_number
+        const { value, category, type } = identityObj
+        const identity = `${value}::${category}::${type}`
+        const maskedIdentity = createMaskedCognitoIdentity(identity)
+        const record = await models.CognitoFlowIdentity.findOne({ where: { maskedIdentity } })
+        if (record) {
+          shouldFail = true
+        } else {
+          const now = Date.now()
+          await models.CognitoFlowIdentity.create({
```

Comment:
we should use a sequelize transaction if this is a separate table from the CognitoFlows. 

https://sequelize.org/master/manual/transactions.html

Reply:
facts